icodebreaker_console/ui.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require './config'
+
+class Ui
+  include UiSupport
+
+  def run
+    loop do
+      puts I18n.t('game.menu')
+      case UiSupport.user_input
+      when I18n.t('command.start') then start
+      when I18n.t('command.rules') then puts I18n.t('game.rules')
+      when I18n.t('command.statistics') then UiSupport.statistics
+      when I18n.t('command.exit') then exit
+      else puts I18n.t('command.error_command') end
+    end
+  end
+
+  def start
+    @codebreaker = IcodebreakerGem::Game.new
+    player_data
+    game
+  end
+
+  def player_data
+    name_enter
+    difficulty_enter
+  end
+
+  def name_enter
+    loop do
+      name = UiSupport.name_question
+      begin
+        @codebreaker.validate_name(name)
+        @codebreaker.name = name
+        break
+      rescue ArgumentError
+        puts I18n.t('game.name_length_error')
```

Comment:
first of all `validate_name` method and all other validations have to return `true/false` statement
I propose to modify validation methods, if successful -> return `true` and we can shorten this `name_enter` method to
```ruby
def name_enter
    loop do
      return true if @codebreaker.validate_name(@codebreaker.name = UiSupport.name_question)
    rescue ArgumentError => e
      puts e.message
    end
end